gpttype_adapter.cpp

```diff
@@ -1095,8 +1096,22 @@ ModelLoadResult gpttype_load_model(const load_model_inputs inputs, FileFormat in
             }
             else
             {
-                float multiplier_rope_base = llamamodel->hparams.rope_freq_base_train/10000.0f;
-                rope_freq_base *= multiplier_rope_base;
+                //float multiplier_rope_base = llamamodel->hparams.rope_freq_base_train/10000.0f;
+                //rope_freq_base *= multiplier_rope_base;
+                
+				//Calculate rope_freq_base using the gradientAI formula, solar requires ctx *8 for correct scaling
+                if(llamamodel->hparams.rope_freq_base_train==10000.0f && file_format_meta.n_tensors==435)
+				{
+					float chi_ctx_train_value = (file_format_meta.n_ctx_train * 8) / (2 * 3.14159265358979323846);
```

Comment:
Actually for the rope calculations you should be adjusting the NTK formula here:
https://github.com/LostRuins/koboldcpp/blob/concedo/gpttype_adapter.cpp#L855

I'm kind of confused, isn't the rope scaling used only based on the selected context size? Why does architecture matter?

Reply:
I did look at that, but it passes an intermediate value of `rope_freq_base` to https://github.com/LostRuins/koboldcpp/blob/1487a4bc812d66654a6c0049bd72e227ba6e4627/gpttype_adapter.cpp#L1101 

By leaving it alone, it retains the logic for not applying auto rope when not needed.